packages/svelte/src/store/index.js

```diff
@@ -46,17 +46,48 @@ export function safe_not_equal(a, b) {
 export function writable(value, start = noop) {
 	/** @type {import('./public').Unsubscriber | null} */
 	let stop = null;
+	let invalidated = false;
 
 	/** @type {Set<import('./private').SubscribeInvalidateTuple<T>>} */
 	const subscribers = new Set();
 
+	function invalidate() {
+		if (invalidated)
```

Comment:
you can use early return pattern
```suggestion
		if (invalidated || !stop) return;
		
			invalidated = true;

			// immediately signal each subscriber as invalid
			for (const subscriber of subscribers) {
				subscriber[1]();
			}
```